text/1946-intra-rustdoc-links.md

```diff
@@ -284,29 +284,36 @@ To be able to link to each item,
 we'll need a way to disambiguate the namespaces.
 Our proposal is this:
 
-- Links to types are written as described earlier,
-  with no pre- or suffix,
-  e.g., `Look at the [FOO] trait`.
-  For consistency,
-  it is also possible to prefix the type with the concrete item type:
-  - Links to `struct`s can be prefixed with `struct `,
-    e.g., `See [struct Foo]`.
-  - Links to `enum`s can be prefixed with `enum `,
-    e.g., `See [enum foo]`.
-  - Links to type aliases can be prefixed with `type `,
-    e.g., `See [type foo]`.
-  - Links to modules can be prefixed with `mod `,
-    e.g., `See [mod foo]`.
+- In unambiguous cases paths can be written as described earlier,
+  with no pre- or suffix, e.g., `Look at the [FOO] trait`. This also
+  applies to modules and tuple structs which exist in both namespaces.
+  Rustdoc will throw an error if you use a non-disambiguated path in
+  the case of there being a value in both the type and value namespace.
+  Non-disambiguated paths cannot be used to link to macros.
+- Links to types can be disambiguated by prefixing them with the concrete
+  item type:
+  - Links to `struct`s can be prefixed with `struct@`,
+    e.g., `See [struct@Foo]`.
+  - Links to `enum`s can be prefixed with `enum@`,
+    e.g., `See [enum@foo]`.
+  - Links to type aliases can be prefixed with `type@`,
+    e.g., `See [type@foo]`.
+  - Links to modules can be prefixed with `mod@`,
+    e.g., `See [mod@foo]`.
```

Comment:
I guess unions and traits should be in this list too?

Reply:
done